de-lib/src/main/java/org/iplantc/de/intercom/client/IntercomFacade.java

```diff
@@ -0,0 +1,70 @@
+package org.iplantc.de.intercom.client;
+
+import org.iplantc.de.shared.DEProperties;
+
+import com.google.web.bindery.autobean.shared.Splittable;
+
+/**
+ * A facade class that manages interaction with Intercom
+ * Created by sriram on 8/1/17.
+ */
+public class IntercomFacade {
```

Comment:
All the methods and fields are `static` (required for JSNI) in this class so I didn't make it Injectable. Also, I found Juice / GIN static injection is not recommended. See - https://github.com/google/guice/wiki/Injections#static-injections